enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+security exposure and resource overhead of running the web console.  I would like a way to install
+a cluster that has no console out of the box, rather than having to disable it post-install or
+modify rendered manifests in a way that requires deep understanding of the OCP components/resources.
+
+* As a team scaffolding a managed service based on openshift, I want to minimize the footprint of my
+clusters to the components I need for the service.
+
+* As a user creating a cluster that will never run an image registry, I do not want the additional overhead
+of running the image registry operator, or have to remove the default registry that is created.
+
+* As a team using openshift for a specific use case such as edge deployments, I want to provide
+an install experience that disables components that aren't needed for my use case.
+
+### API Extensions
+
+N/A - no api extensions are being introduced in this proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+The CVO already has the ability to respect annotations on resources, as can be seen
+[here](https://github.com/openshift/cluster-kube-apiserver-operator/blob/c03c9edf5fddf4e3fb1bc6d7afcd2a2284ca03d8/manifests/0000_20_kube-apiserver-operator_06_deployment.yaml#L10) and leveraged [here](https://github.com/openshift/hypershift/blob/b98bc43467a73921891ede7d6477757b647f1985/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L152-L153).
+This proposal consists of two parts:
+
+1) Formalizing a concept of a "capability" annotation which allows a given resource to be excluded based
+on installer input. For example the console related resources could be annotated as
+
+```yaml
+annotations:
+  capability.openshift.io/name: "console"
+```
+
+The `oc adm release new` command will be updated to extract these annotations and embed a list of the valid
+capabilities in the release payload image.  This list can be consulted by users to determine the set of
+capabilities they can enable/disable when configuring the installation.
+
+Note: Resources are enabled unless explicitly excluded by configuration.  In the future we may allow
+resources to be excluded unless explicitly included, this can be done by introducing a new annotation
+such as `capability.openshift.io/defaultDisable`.  If we do so, we will need to define how that choice
+interacts with the cluster scoped "default capabilities on/off" configuration setting.  For now, this
+EP takes the position that individual components that exist in the payload cannot choose to be
+default-disabled.
+
+2) Defining an [install config api](https://github.com/openshift/installer/blob/790048166067273d34f76bea4220fa395b1cce1b/pkg/types/installconfig.go#L70) field whereby the user can opt out of specific capabilities.
+
+```yaml
+InstallConfig.capabilities
+  inclusionDefault: [Include|Exclude]
+  exclude:
+  - console
+  - samples
+  include:
+  - registry
+```
```

Comment:
We should break this example in to two example i.e. one with `inclusionDefault: Exclude` and another with `inclusionDefault: Include`. I think we will not support both at the same time.

Reply:
ack, will update